### PR TITLE
Feature/sanitize html front aswell

### PR DESCRIPTION
**Pull request by mjvkeulen**
## Description
Sandboxed all v-html to iframes to block all script execution.

### CODE REVIEW

File: src/vue/src/components/assets/SandboxedIframe.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        marginwidth="0"
+        marginheight="0"
+        scrolling="no"
+        class="w-100 theme-iframe"
+    />
+</template>
+
+<script>
+export default {
+    name: 'Test',
```
Comment (engelhamer) [inline src/vue/src/components/assets/SandboxedIframe.vue]: This should be changed.

File: src/vue/src/components/entry/EntryFields.vue [hunk 2]
```diff
@@ -145,7 +146,8 @@ export default {
         'url-input': urlInput,
         'pdf-display': pdfDisplay,
         'file-download-button': fileDownloadButton,
-        'image-file-display': imageFileDisplay
+        'image-file-display': imageFileDisplay,
+        sandboxedIframe
```
Comment (engelhamer) [inline src/vue/src/components/entry/EntryFields.vue]: Still surprises me that this works straight away.

File: src/vue/src/main.js [hunk 3]
```diff
@@ -8,58 +8,7 @@ import '../node_modules/bootstrap/dist/css/bootstrap.css'
...
-import 'vue-awesome/icons/long-arrow-down'
-import 'vue-awesome/icons/comments'
-import 'vue-awesome/icons/cog'
-import 'vue-awesome/icons/clock-o'
-import 'vue-awesome/icons/print'
-import 'vue-awesome/icons/github'
-import 'vue-awesome/icons/linkedin'
-import 'vue-awesome/icons/file'
-import 'vue-awesome/icons/search'
+import '@/helpers/vue-awesome-icons.js'
```
Comment (engelhamer) [inline src/vue/src/main.js]: Smart.

File: src/vue/src/components/journal/CommentCard.vue [hunk 4]
```diff
@@ -17,7 +17,7 @@
                             <icon name="edit"/>
                             Edit
                         </b-button>
-                        <div v-html="comment.text"/>
+                        <sandboxed-iframe :content="comment.text"/>
```
Comment (engelhamer) [inline src/vue/src/components/journal/CommentCard.vue]: This introduces some peculiar padding: The iframce clears for the buttons floating on the right.

screenshot from 2018-11-26 16-43-28

Comment (engelhamer) [reply src/vue/src/components/journal/CommentCard.vue]: Nice!

**Comment by engelhamer** [comment]
Changed the colors to match the hourglass and timestamp.

screenshot from 2018-11-28 17-36-16

File: src/vue/src/components/journal/CommentCard.vue (hunk 4 above)
Comment (engelhamer) [reply src/vue/src/components/journal/CommentCard.vue]: Changed the colors to match the hourglass and timestamp.

screenshot from 2018-11-28 17-36-16

File: src/vue/src/Header.vue [hunk 5]
```diff
@@ -19,8 +19,14 @@
 
         <b-collapse is-nav id="nav-collapse">
             <b-navbar-nav class="mr-auto">
-                <b-nav-item :to="{ name : 'Home' }">Courses</b-nav-item>
-                <b-nav-item :to="{ name : 'AssignmentsOverview' }">Assignments</b-nav-item>
+                <b-nav-item :to="{ name : 'Home' }">
+                    <icon name="home"/>
+                    Courses
```
Comment (Larspolo) [inline src/vue/src/Header.vue]: Courses stay lit up when going inside a course, and it also switched back top courses being lit when going from assignments overview to a different page. Can this be fixed in this PR or maybe just create another issue?

File: src/vue/src/components/assets/SandboxedIframe.vue [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        setCustomStyle (obj) {
+            var doc = obj.contentWindow.document
+
+            var css = `
+body {
+    font-family: 'Roboto', sans-serif;
+    text-decoration: none;
+    color: #252C39;
+}`
```
Comment (Larspolo) [inline src/vue/src/components/assets/SandboxedIframe.vue]: Maybe add link style as well?
Now:
image

Before:
image

File: src/vue/src/components/assets/SandboxedIframe.vue [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+            var style = document.createElement('style')
+            style.type = 'text/css'
+            style.appendChild(document.createTextNode(css))
```
Comment (Larspolo) [inline src/vue/src/components/assets/SandboxedIframe.vue]: Cant this also be done by just using an url to a stylesheet? I would prefer that over typing all the css in here

File: src/vue/src/Header.vue (hunk 5 above)
Comment (engelhamer) [reply src/vue/src/Header.vue]: That is desired behaviour, and has always been the case. Maybe we could highlight 'Assignments' once the user is in an assignment, that would be nice!

Comment (engelhamer) [reply src/vue/src/Header.vue]: Doesn't have anything tot do with sanitization (actually my previous addition did nog either, sorry for that!), so another issue would be appropriate.

File: src/vue/src/components/assets/SandboxedIframe.vue (hunk 7 above)
Comment (engelhamer) [reply src/vue/src/components/assets/SandboxedIframe.vue]: That would be nice, tackles a lot of issues at once (the current implementation obviously doesn't scale).

Comment (mjvkeulen) [reply src/vue/src/components/assets/SandboxedIframe.vue]: Could use a stylesheet, if all we want to do is change link styling I would rather just append.

Comment (mjvkeulen) [reply src/vue/src/components/assets/SandboxedIframe.vue]: Using theme blue

image

**Pull request merged**